apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -127,28 +176,58 @@ int stack_prot_main(int argc, char *argv[])
 			ch = getchar();
 			i = 0;
 			if (ch == 'N' || ch == 'n') {
-				stack_prot_inlimit();
-				printf(" Test stack access within range using array size - Passed !!\n");
+				if (type_kthread) {
+					ret_chk = ioctl(tc_fd, TESTIOC_KTHREAD_STACK_PROTECTION_TEST, (unsigned long)&ch);
+					if (ret_chk < 0) {
+						printf("ERROR: ioctl() failed with errno: %d\n", errno);
+						close(tc_fd);
+						return ERROR;
+					}
+				} else {
+					stack_prot_inlimit();
+					printf(" Test stack access within range using array size - Passed !!\n");
+				}
 			} else if (ch == 'O' || ch == 'o') {
 				printf("\nPress A - Test Overflow using array declaration\n");
 				printf("Press F - Test Overflow using recursive function\n");
 				ch = getchar();
 				if (ch == 'A' || ch == 'a') {
-					stack_overflow_array();
-					printf(" Test stack access in overflow - Failed !!\n");
+					if (type_kthread) {
+						ret_chk = ioctl(tc_fd, TESTIOC_KTHREAD_STACK_PROTECTION_TEST, (unsigned long)&ch);
+						if (ret_chk < 0) {
+							printf("ERROR: ioctl() failed with errno: %d\n", errno);
+							close(tc_fd);
+							return ERROR;
+						}
+					} else {
+						stack_overflow_array();
+						printf(" Test stack access in overflow - Failed !!\n");
+					}
 				} else if (ch == 'F' || ch == 'f') {
-					stack_overflow_func(&i);
-					printf(" Test stack access in overflow - Failed !!\n");
+					if (type_kthread) {
+						ret_chk = ioctl(tc_fd, TESTIOC_KTHREAD_STACK_PROTECTION_TEST, (unsigned long)&ch);
+						if (ret_chk < 0) {
+							printf("ERROR: ioctl() failed with errno: %d\n", errno);
+							close(tc_fd);
+							return ERROR;
+						}
+					} else {
+						stack_overflow_func(&i);
+						printf(" Test stack access in overflow - Failed !!\n");
+					}
 				}
 			} else if (ch == 'E' || ch == 'e') {
 				printf(" Exit the Test ..\n");
+				close(tc_fd);
```

Comment:
1. Need to check type_kthread is true. If you don't, this will try to close with some random value.
  Test scenario is testing user thread once and exiting.
2. Need to close opened fd after some complex test.
  Test scenario is testing kernel thread, user thread and exiting.

Reply:
1.) Added check for type_kthread as per your suggestion. Tested using mentioned scenario.
2.) Right now, in one go, we can test either kernel thread or user thread but not both.
Need to exit test and run again for testing other thread type.
So, can't replicate the complex test you have mentioned.